feat: add form deletion

dodumosu opened the pull request:
this PR adds the ability to delete forms.

see #569

dodumosu committed:
feat: add deletion for forms

see: #569

dodumosu committed:
fix: specify button type

takinbo commented on apollo/formsframework/views_forms.py:
```diff
@@ -429,3 +433,14 @@ def import_form_schema():
     return redirect(url_for('formsview.index'))
+
+
+@route(bp, '/delete', methods=['POST'])
```
@dodumosu all routes are now being handled within `views_admin.py`. Maybe you could use the same format for all the other views there?

dodumosu committed:
chore: move form deletion to form admin view group

dodumosu committed:
chore: update template with view name

- update template with view name
- fix broken JS
- disable import button once import form is submitted

takinbo commented:
PR works most of the time but occasionally, this happens:

![screenshot-1572872817](https://user-images.githubusercontent.com/41004/68123154-15ab2300-ff04-11e9-8b13-df256be928a5.png)

takinbo reviewed (changes requested):
PR needs some bug fixing to be merged.

dodumosu committed:
fix: submit the delete form always

takinbo commented on apollo/templates/admin/form_list.html:
```diff
@@ -177,6 +189,21 @@ <h5 id="formImportModalLabel" class="modal-title">{{ _('Import Form') }}</h5>
+    // and the target is, well, the icon. so we have to go up one more
+    // level if that's the case
+    var form = event.target.localName.toLowerCase() === 'button' ? event.target.parentElement : event.target.parentElement.parentElement;
+    var result = confirm("{{ _('Are you sure you want to delete this form? All submissions for it will be deleted too!') }}");
```
Can we rephrase to `"All data linked to this form will be deleted too."`?

dodumosu committed:
chore: change prompt for form deletion

this commit contains the following changes:
- change prompt on form deletion
- remove Apollo 2 submission update task
- refactor form deletion so linked submissions are deleted explicitly
with the form in a background task

takinbo reviewed (changes requested):
I'm still unable to delete forms. See this [video](https://youtu.be/MPCclS4HJuM) for a demonstration.

dodumosu commented:
was the worker process running when you used it? the actual deletion is done in a worker thread, because it deletes all submissions first.

takinbo commented:
> was the worker process running when you used it? the actual deletion is done in a worker thread, because it deletes all submissions first.

Yes the worker thread was running and there were no submissions.

dodumosu commented:
that's definitely weird, because i tried it out and didn't have any issues: https://vimeo.com/510580040
